opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -178,11 +180,13 @@ def create(
         resource = _DEFAULT_RESOURCE
 
         otel_experimental_resource_detectors = environ.get(
-            OTEL_EXPERIMENTAL_RESOURCE_DETECTORS, "otel"
+            OTEL_EXPERIMENTAL_RESOURCE_DETECTORS, "otel,os"
         ).split(",")
 
         if "otel" not in otel_experimental_resource_detectors:
             otel_experimental_resource_detectors.append("otel")
+        if "os" not in otel_experimental_resource_detectors:
```

Comment:
`otel_experimental_resource_detectors` should not be a list but a `set` because all its elements are unique and they are not ordered:

```suggestion
        otel_experimental_resource_detectors = {"otel", "os"}.union(
            set(
                [
                    otel_experimental_resource_detector.strip()
                    for otel_experimental_resource_detector
                    in environ.get("OTEL_EXPERIMENTAL_RESOURCE_DETECTORS", "").
                    split(",")
                    if otel_experimental_resource_detector
                ]
            )
        )
```

Reply:
That's a good suggestion, thank you. [Put you in](https://github.com/open-telemetry/opentelemetry-python/pull/3992/commits/a32a6058ad13dd7d432c8629d87dfdb6f8cc074c) as a `Co-Authored-By` with a value from the repo's git logs.